sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -468,6 +468,15 @@ object GpuOverrides extends Logging {
     case _ => false
   }
 
+  private[this] def isArrayOfStructType(dataType: DataType) = dataType match {
+    case ArrayType(elementType, _) =>
+      elementType match {
+        case StructType(_) => true
+        case _ => false
+      }
+    case _ => false
+  }
```

Comment:
This PR enables single level array so I think we should also not support array of array:
```suggestion
  private[this] def isArrayOfStructType(dataType: DataType) = dataType match {
    case ArrayType(elementType, _) =>
      elementType match {
        case StructType(_) |  ArrayType => true
        case _ => false
      }
    case _ => false
  }
```

Reply:
Note that the type checks below also need to be updated.

Reply:
Thank you! Currently `order by` is falling back to cpu on nested arrays, integration test is `test_single_orderby_fallback_for_multilevel_array`. I am not sure if we still need to update this function.

Reply:
If all the current tests pass after updating this function then I think we should.
But this is not very important since we are going to have `order by` support on nested arrays in the next PR.